sign/schnorr.go

```diff
@@ -0,0 +1,95 @@
+package sign
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/random"
+)
+
+// HashFunc is the hash function used to hash the message for the signature
+// generation and verification
+var HashFunc = sha256.New
+
+// Schnorr creates a Schnorr signature from a msg and a private key. This
+// signature can be verified with VerifySchnorr. It's also a valid EdDSA
+// signature.
+func Schnorr(suite abstract.Suite, private abstract.Scalar, msg []byte) ([]byte, error) {
+	// using notation from https://en.wikipedia.org/wiki/Schnorr_signature
+	// create random secret k and public point commitment r
+	k := suite.Scalar().Pick(random.Stream)
+	r := suite.Point().Mul(nil, k)
+
+	// create challenge e based on message and r
+	public := suite.Point().Mul(nil, private)
+	e, err := hash(suite, public, r, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// compute response s = k - x*e
+	xe := suite.Scalar().Mul(private, e)
+	s := suite.Scalar().Sub(k, xe)
+
+	var b bytes.Buffer
+	if _, err := e.MarshalTo(&b); err != nil {
+		return nil, err
+	}
+	if _, err := s.MarshalTo(&b); err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+// VerifySchnorr verifies a given Schnorr signature. It returns nil iff the
```

Comment:
Exactly, that's what I meant with this `iff`: it returns nil if and only if the signature is valid.